solidity/test/zeto_anon_nullifier_kyc.ts

```diff
@@ -0,0 +1,436 @@
+// Copyright © 2024 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ethers, ignition } from 'hardhat';
+import { ContractTransactionReceipt, Signer, BigNumberish, AddressLike } from 'ethers';
+import { expect } from 'chai';
+import { loadCircuit, Poseidon, encodeProof, kycHash } from "zeto-js";
+import { groth16 } from 'snarkjs';
+import { Merkletree, InMemoryDB, str2Bytes } from '@iden3/js-merkletree';
+import zetoModule from '../ignition/modules/zeto_anon_nullifier_kyc';
+import erc20Module from '../ignition/modules/erc20';
+import { UTXO, User, newUser, newUTXO, newNullifier, doMint, ZERO_UTXO, parseUTXOEvents } from './lib/utils';
+import { loadProvingKeys, prepareDepositProof, prepareNullifierWithdrawProof } from './utils';
+
+describe("Zeto based fungible token with anonymity, KYC, using nullifiers without encryption", function () {
```

Comment:
Should there be a test for unregistered user checks?

Reply:
covered in https://github.com/hyperledger-labs/zeto/pull/25/files#diff-16c442cf7ed2d5adf832bff0e695df6a69fd1d71081cb25cad880fc93cd9ca79R146

Reply:
added two new tests for the above suggested cases